mountpoint-s3/src/prefetch.rs

```diff
@@ -246,7 +246,52 @@ where
             next_seq_offset = self.next_sequential_read_offset,
             "read"
         );
+        let result = self.try_read(offset, length).await;
+        if result.is_err() {
+            self.reset_prefetch_to_offset(offset);
```

Comment:
Just wondering if there are any other errors (apart from `PrefetchReadError` on `try_seek`) that will trigger reset of the prefetcher after this change. But even if we don't know this set exactly, I think, resetting is a safe choice to make. I can not imagine errors which would allow further reading from the task.

Reply:
Good point! 

There is only one case: failing integrity check when extending the response buffer. We were previously canceling the tasks, but not fully resetting the prefetcher state. My latest commit fixes that.